tools/version-bump/lib/update.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { readFile, writeFile, stat } from 'fs/promises';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from './logger';
+
+/**
+ * Update plugin readme stable tag.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeStableTag = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/Stable tag: \d.\d.\d\n/m,
+			`Stable tag: ${ nextVersion }\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme stable tag' );
+	}
+};
+
+/**
+ * Update plugin readme changelog.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeChangelog = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/= \d.\d.\d \d\d\d\d-XX-XX =\n/m,
+			`= ${ nextVersion } ${ new Date().getFullYear() }-XX-XX =\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme changelog' );
+	}
+};
+
+/**
+ * Update plugin class file.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateClassPluginFile = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/includes/class-${ plugin }.php`;
+
+	try {
+		await stat( filePath );
+	} catch ( e ) {
+		// Class file does not exist, return early.
+		return;
+	}
+
+	try {
+		const classPluginFileContents = await readFile( filePath, 'utf8' );
+
+		const updatedClassPluginFileContents = classPluginFileContents.replace(
+			/public \$version = '\d.\d.\d';\n/m,
+			`public $version = '${ nextVersion }';\n`
+		);
+
+		await writeFile( filePath, updatedClassPluginFileContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update plugin file.' );
+	}
+};
+
+/**
+ * Update plugin JSON files.
+ *
+ * @param {string} type        plugin to update
+ * @param {string} plugin      plugin to update
+ * @param {string} nextVersion version to bump to
+ */
+export const updateJSON = async (
+	type: 'package' | 'composer',
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/${ type }.json`;
+	try {
+		const composerJson = JSON.parse( await readFile( filePath, 'utf8' ) );
```

Comment:
Is this a safe path? shouldn't this be absolute path from the top of `woocommerce` or something? This would only work if you run from the correct directory otherwise

Reply:
I updated all the paths to be more robust in https://github.com/woocommerce/woocommerce/pull/34555/commits/8c4358e4d37426203c36e9582aef910448d20dbc